tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -300,66 +300,104 @@ void check_vectors(const Variables<T1>& t1, const Variables<T2>& t2) {
+
+  check_vectors(test_variable_type{s, 4. * vars_val},
+                vars + vars + vars + vars);
+  check_vectors(test_variable_type{s, 3. * vars_val}, vars + (vars + vars));
+  // clang-tidy: point sides of overloaded operator are equivalent
+  check_vectors(test_variable_type{s, -2. * vars_val},
+                vars - vars - vars - vars);  // NOLINT
+  check_vectors(test_variable_type{s, vars_val},
+                vars - (vars - vars));  // NOLINT
+  check_vectors(test_variable_type{s, -vars_val}, vars - (vars + vars));
+  check_vectors(test_variable_type{s, vars_val}, vars - vars + vars);  // NOLINT
+  check_vectors(test_variable_type{s, vars_val}, vars + vars - vars);
+
+  test_variable_type test_assignment(vars * rvs[0]);
+  auto expected = vars_val * rvs[0];
+  test_assignment += test_variable_type{s, vars_val};
+  expected += vars_val;
+  check_vectors(test_variable_type{s, expected}, test_assignment);
+  test_assignment -= test_variable_type{s, rvs[1]};
+  expected -= rvs[1];
+  check_vectors(test_variable_type{s, expected}, test_assignment);
+  test_assignment *= rvs[2];
+  expected *= rvs[2];
+  check_vectors(test_variable_type{s, expected}, test_assignment);
+  test_assignment /= rvs[0];
+  expected /= rvs[0];
+  check_vectors(test_variable_type{s, expected}, test_assignment);
+
+  test_assignment += test_variable_type{s, vars_val} * rvs[1];
+  expected += vars_val * rvs[1];
+  check_vectors(test_variable_type{s, expected}, test_assignment);
+  test_assignment -= test_variable_type{s, rvs[2]} * rvs[0];
+  expected -= rvs[2] * rvs[0];
+  check_vectors(test_variable_type{s, expected}, test_assignment);
+
+  test_variable_type test_assignment2{s};
   test_assignment2 = test_assignment * 1.0;
   CHECK(test_assignment2 == test_assignment);
 
   const auto check_components = [](const auto& variables,
-                                   const DataVector& datavector) {
+                                   const VectorType& vector) {
     tmpl::for_each<typename std::decay_t<decltype(variables)>::tags_list>(
-        [&variables, &datavector](auto tag) {
+        [&variables, &vector](auto tag) {
           using Tag = tmpl::type_from<decltype(tag)>;
           for (const auto& component : get<Tag>(variables)) {
-            CHECK(component == datavector);
+            CHECK(component == vector);
           }
         });
   };
-  const DataVector dv{1., 2., 3., 4.};
-  test_variable_type test_datavector_math(4, 2.);
-  test_datavector_math *= dv;
-  check_components(test_datavector_math, {2., 4., 6., 8.});
-  check_components(test_datavector_math * dv, {2., 8., 18., 32.});
-  check_components(dv * test_datavector_math, {2., 8., 18., 32.});
-  test_datavector_math *= dv;
-  check_components(test_datavector_math, {2., 8., 18., 32.});
-  check_components(test_datavector_math / dv, {2., 4., 6., 8.});
-  test_datavector_math /= dv;
-  check_components(test_datavector_math, {2., 4., 6., 8.});
+
+  VectorType vec{4};
```

Comment:
Could you add a comment about what the below tests are doing compared to the above tests? Like what I had suggested on a previous.